egs/librispeech/ASR/local/prepare_lang.py

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/env python3
+
+# Copyright (c)  2021  Xiaomi Corporation (authors: Fangjun Kuang)
+
+"""
+This script takes as input a lexicon file "data/lang/lexicon.txt"
+consisting of words and phones and does the following:
+
+1. Add disambiguation symbols to the lexicon and generate lexicon_disambig.txt
+
+2. Generate phones.txt, the phones table mapping a phone to a unique integer.
+
+3. Generate words.txt, the words table mapping a word to a unique integer.
+
+4. Generate L.pt, in k2 format. It can be loaded by
+
+        d = torch.load("L.pt")
+        lexicon = k2.Fsa.from_dict(d)
+
+5. Generate L_disambig.pt, in k2 format.
+"""
+import math
+import re
+import sys
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+import k2
+import torch
+
+Lexicon = List[Tuple[str, List[str]]]
+
+
+def read_lexicon(filename: str) -> Lexicon:
```

Comment:
It’s confusing because we already have a `Lexicon` class elsewhere; do you think this function could be a constructor-method on that class? Technically you can derive all of L.fst, phones.txt, and words.txt from lexicon[p].txt.

Reply:
After reading the whole file, I think we should either stick to the `Lexicon` class and move most of these functions there, or remove the `Lexicon` class altogether, these are clearly very overlapping.